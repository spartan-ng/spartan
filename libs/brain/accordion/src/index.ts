import { NgModule } from '@angular/core';

import { BrnAccordion, BrnAccordionItem, BrnAccordionTrigger } from './lib/brn-accordion';
import { BrnAccordionContent } from './lib/brn-accordion-content';

export * from './lib/brn-accordion';
export * from './lib/brn-accordion-content';

export const BrnAccordionImports = [BrnAccordion, BrnAccordionContent, BrnAccordionItem, BrnAccordionTrigger] as const;

@NgModule({
	imports: [...BrnAccordionImports],
	exports: [...BrnAccordionImports],
})
export class BrnAccordionModule {}
