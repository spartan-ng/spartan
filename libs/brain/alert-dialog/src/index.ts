import { NgModule } from '@angular/core';

import { BrnAlertDialog } from './lib/brn-alert-dialog';
import { BrnAlertDialogContent } from './lib/brn-alert-dialog-content';
import { BrnAlertDialogDescription } from './lib/brn-alert-dialog-description';
import { BrnAlertDialogOverlay } from './lib/brn-alert-dialog-overlay';
import { BrnAlertDialogTitle } from './lib/brn-alert-dialog-title';
import { BrnAlertDialogTrigger } from './lib/brn-alert-dialog-trigger';

export * from './lib/brn-alert-dialog';
export * from './lib/brn-alert-dialog-content';
export * from './lib/brn-alert-dialog-description';
export * from './lib/brn-alert-dialog-overlay';
export * from './lib/brn-alert-dialog-title';
export * from './lib/brn-alert-dialog-trigger';

export const BrnAlertDialogImports = [
	BrnAlertDialog,
	BrnAlertDialogOverlay,
	BrnAlertDialogTrigger,
	BrnAlertDialogContent,
	BrnAlertDialogTitle,
	BrnAlertDialogDescription,
] as const;

@NgModule({
	imports: [...BrnAlertDialogImports],
	exports: [...BrnAlertDialogImports],
})
export class BrnAlertDialogModule {}
