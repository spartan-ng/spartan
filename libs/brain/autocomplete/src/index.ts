import { NgModule } from '@angular/core';
import { BrnAutocomplete } from './lib/brn-autocomplete';
import { BrnAutocompleteEmpty } from './lib/brn-autocomplete-empty';
import { BrnAutocompleteGroup } from './lib/brn-autocomplete-group';
import { BrnAutocompleteItem } from './lib/brn-autocomplete-item';
import { BrnAutocompleteList } from './lib/brn-autocomplete-list';
import { BrnAutocompleteSearchInput } from './lib/brn-autocomplete-search-input';

export * from './lib/brn-autocomplete';
export * from './lib/brn-autocomplete-empty';
export * from './lib/brn-autocomplete-group';
export * from './lib/brn-autocomplete-item';
export * from './lib/brn-autocomplete-item.token';
export * from './lib/brn-autocomplete-list';
export * from './lib/brn-autocomplete-search-input';
export * from './lib/brn-autocomplete-search-input.token';
export * from './lib/brn-autocomplete.token';

export const BrnAutocompleteImports = [
	BrnAutocomplete,
	BrnAutocompleteEmpty,
	BrnAutocompleteGroup,
	BrnAutocompleteItem,
	BrnAutocompleteList,
	BrnAutocompleteSearchInput,
] as const;

@NgModule({
	imports: [...BrnAutocompleteImports],
	exports: [...BrnAutocompleteImports],
})
export class BrnAutocompleteModule {}
