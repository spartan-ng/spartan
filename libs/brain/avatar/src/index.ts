import { NgModule } from '@angular/core';

import { BrnAvatar } from './lib/brn-avatar';
import { BrnAvatarFallback } from './lib/fallback';
import { BrnAvatarImage } from './lib/image';

export * from './lib/brn-avatar';
export * from './lib/fallback';
export * from './lib/image';
export * from './lib/util';

export const BrnAvatarImports = [BrnAvatar, BrnAvatarFallback, BrnAvatarImage] as const;

@NgModule({
	imports: [...BrnAvatarImports],
	exports: [...BrnAvatarImports],
})
export class BrnAvatarModule {}
