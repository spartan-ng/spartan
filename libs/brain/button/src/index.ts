import { NgModule } from '@angular/core';
import { BrnButton } from './lib/brn-button';

export * from './lib/brn-button';

@NgModule({
	imports: [BrnButton],
	exports: [BrnButton],
})
export class BrnButtonModule {}
