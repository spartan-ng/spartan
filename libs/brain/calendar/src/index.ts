export * from './lib/brn-calendar';
export * from './lib/brn-calendar-cell';
export * from './lib/brn-calendar-cell-button';
export * from './lib/brn-calendar-grid';
export * from './lib/brn-calendar-header';
export * from './lib/brn-calendar-next-button';
export * from './lib/brn-calendar-previous-button';
export * from './lib/brn-calendar-week';
export * from './lib/brn-calendar-weekday';
export * from './lib/brn-calendar.token';
export * from './lib/i18n/calendar-i18n';
export * from './lib/mode/brn-calendar-multiple';
export * from './lib/mode/brn-calendar-range';

import { NgModule } from '@angular/core';
import { BrnCalendar } from './lib/brn-calendar';
import { BrnCalendarCell } from './lib/brn-calendar-cell';
import { BrnCalendarCellButton } from './lib/brn-calendar-cell-button';
import { BrnCalendarGrid } from './lib/brn-calendar-grid';
import { BrnCalendarHeader } from './lib/brn-calendar-header';
import { BrnCalendarNextButton } from './lib/brn-calendar-next-button';
import { BrnCalendarPreviousButton } from './lib/brn-calendar-previous-button';
import { BrnCalendarWeek } from './lib/brn-calendar-week';
import { BrnCalendarWeekday } from './lib/brn-calendar-weekday';
import { BrnCalendarMulti } from './lib/mode/brn-calendar-multiple';
import { BrnCalendarRange } from './lib/mode/brn-calendar-range';

export const BrnCalendarImports = [
	BrnCalendarCellButton,
	BrnCalendarGrid,
	BrnCalendarHeader,
	BrnCalendarNextButton,
	BrnCalendarPreviousButton,
	BrnCalendarWeek,
	BrnCalendarWeekday,
	BrnCalendar,
	BrnCalendarCell,
	BrnCalendarMulti,
	BrnCalendarRange,
] as const;

@NgModule({
	imports: [...BrnCalendarImports],
	exports: [...BrnCalendarImports],
})
export class BrnCalendarModule {}
