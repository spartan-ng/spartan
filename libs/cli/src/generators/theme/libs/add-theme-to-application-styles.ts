// All credit goes to the incredible folks at Nx who use this code to update the app styles when adding tailwind
// Check out the code here: https://github.com/nrwl/nx/blob/master/packages/angular/src/generators/setup-tailwind/lib/update-application-styles.ts

import { type ProjectConfiguration, type Tree, joinPathFragments, readJson, stripIndents } from '@nx/devkit';
import { type PackageJson } from 'nx/src/utils/package-json';
import * as semver from 'semver';
import { ThemeName, themes } from './colors';

export interface AddThemeToApplicationStylesOptions {
	project: string;
	theme: ThemeName;
	addCdkStyles: boolean;
	stylesEntryPoint?: string;
	prefix?: string;
}

export function addThemeToApplicationStyles(
	tree: Tree,
	options: AddThemeToApplicationStylesOptions,
	project: ProjectConfiguration,
): void {
	const packageJson = readJson<PackageJson>(tree, 'package.json');

	let tailwindVersion = 3;

	if ('tailwindcss' in packageJson.devDependencies) {
		const version = packageJson.devDependencies['tailwindcss'];
		tailwindVersion = semver.coerce(version)?.major ?? 3;
	}

	const tailwindImport = tailwindVersion === 4 ? '@import "@spartan-ng/brain/hlm-tailwind-preset.css";' : '';

	const prefix = options.prefix ? ` .${options.prefix}` : '';
	let stylesEntryPoint = options.stylesEntryPoint;

	if (stylesEntryPoint && !tree.exists(stylesEntryPoint)) {
		throw new Error(`The provided styles entry point "${stylesEntryPoint}" could not be found.`);
	}

	if (!stylesEntryPoint) {
		stylesEntryPoint = findStylesEntryPoint(tree, project);

		if (!stylesEntryPoint) {
			throw new Error(
				stripIndents`Could not find a styles entry point for project "${options.project}".
        Please specify a styles entry point using the "stylesEntryPoint" option.`,
			);
		}
	}

	const stylesEntryPointContent = tree.read(stylesEntryPoint, 'utf-8');

	const CDK_IMPORT = `@import '@angular/cdk/overlay-prebuilt.css';`;
	const ckdOverlayImport = stylesEntryPointContent.includes(CDK_IMPORT) ? '' : CDK_IMPORT;

	const rootFontSans = stylesEntryPointContent.includes('--font-sans')
		? ''
		: `:root {
     --font-sans: ''
     }`;

	const colorScheme = tailwindVersion === 4 ? themes[options.theme].cssVarsV4 : themes[options.theme].cssVarsV3;

	tree.write(
		stylesEntryPoint,
		stripIndents`
    ${ckdOverlayImport}

    ${stylesEntryPointContent}
    ${tailwindImport}

    ${rootFontSans}

		:root${prefix} {
			${Object.entries(colorScheme.light)
				.map(([key, value]) => `--${key}: ${value};`)
				.join('\n')}
		}

		:root.dark${prefix} {
			${Object.entries(colorScheme.dark)
				.map(([key, value]) => `--${key}: ${value};`)
				.join('\n')}
		}

    @layer base {
      * {
        @apply border-border;
      }
    }`,
	);
}

function findStylesEntryPoint(tree: Tree, project: ProjectConfiguration): string | undefined {
	// first check for common names
	const possibleStylesEntryPoints = [
		joinPathFragments(project.sourceRoot ?? project.root, 'styles.css'),
		joinPathFragments(project.sourceRoot ?? project.root, 'styles.scss'),
		joinPathFragments(project.sourceRoot ?? project.root, 'styles.sass'),
		joinPathFragments(project.sourceRoot ?? project.root, 'styles.less'),
	];

	const stylesEntryPoint = possibleStylesEntryPoints.find((s) => tree.exists(s));
	if (stylesEntryPoint) {
		return stylesEntryPoint;
	}

	// then check for the specified styles in the build configuration if it exists
	const styles: Array<string | { input: string; inject: boolean }> = project.targets?.build.options?.styles;

	if (!styles) {
		return undefined;
	}

	// find the first style that belongs to the project source
	const style = styles.find((s) =>
		typeof s === 'string'
			? s.startsWith(project.root) && tree.exists(s)
			: s.input.startsWith(project.root) && s.inject !== false && tree.exists(s.input),
	);

	if (!style) {
		return undefined;
	}

	return typeof style === 'string' ? style : style.input;
}
