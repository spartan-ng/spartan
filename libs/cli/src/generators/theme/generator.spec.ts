import { applicationGenerator, E2eTestRunner, setupTailwindGenerator, UnitTestRunner } from '@nx/angular/generators';
import { readProjectConfiguration, Tree, updateJson } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from 'nx/src/devkit-testing-exports';
import { addThemeToApplicationStyles } from './libs/add-theme-to-application-styles';

jest.mock('@nx/devkit', () => {
	const original = jest.requireActual('@nx/devkit');
	return {
		...original,
		// eslint-disable-next-line @typescript-eslint/no-empty-function
		ensurePackage: () => {},
		createProjectGraphAsync: jest.fn().mockResolvedValue({
			nodes: {},
			dependencies: {},
		}),
	};
});

describe('Theme generator', () => {
	let tree: Tree;

	beforeEach(async () => {
		tree = createTreeWithEmptyWorkspace();
		await applicationGenerator(tree, {
			directory: 'website',
			e2eTestRunner: E2eTestRunner.None,
			linter: 'none',
			name: 'website',
			skipFormat: true,
			unitTestRunner: UnitTestRunner.None,
			skipTests: true,
		});
	});

	describe('Tailwind v3', () => {
		beforeEach(async () => {
			await setupTailwindGenerator(tree, {
				project: 'website',
				skipFormat: true,
			});
		});

		it('should not add the Tailwind 4 preset global stylesheet', async () => {
			const project = readProjectConfiguration(tree, 'website');

			addThemeToApplicationStyles(
				tree,
				{
					theme: 'zinc',
					project: 'website',
					radius: 2,
					addCdkStyles: true,
				},
				project,
			);

			const styles = tree.read('website/src/styles.css', 'utf8');

			expect(styles).not.toContain('@import "@spartan-ng/brain/hlm-tailwind-preset.css";');
		});
	});

	describe('Tailwind v4', () => {
		beforeEach(async () => {
			updateJson(tree, 'package.json', (json) => {
				json.devDependencies = {
					...json.devDependencies,
					tailwindcss: '^4.0.0',
				};
				return json;
			});

			tree.write('website/src/styles.css', '@import "tailwindcss";');
		});

		it('should add the Tailwind 4 preset global stylesheet', async () => {
			const project = readProjectConfiguration(tree, 'website');

			addThemeToApplicationStyles(
				tree,
				{
					theme: 'zinc',
					project: 'website',
					radius: 2,
					addCdkStyles: true,
				},
				project,
			);

			const styles = tree.read('website/src/styles.css', 'utf8');

			expect(styles).toContain('@import "@spartan-ng/brain/hlm-tailwind-preset.css";');
		});
	});
});
