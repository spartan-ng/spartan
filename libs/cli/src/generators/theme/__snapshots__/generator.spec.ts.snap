// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme generator should add the gray theme styles to the global stylesheet 1`] = `
"@import "@angular/cdk/overlay-prebuilt.css";

@import "tailwindcss";
@import "@spartan-ng/brain/hlm-tailwind-preset.css";

:root {
color-scheme: light;

--font-sans: '';

--background: oklch(1 0 0);
--foreground: oklch(0.13 0.028 261.692);
--card: oklch(1 0 0);
--card-foreground: oklch(0.13 0.028 261.692);
--popover: oklch(1 0 0);
--popover-foreground: oklch(0.13 0.028 261.692);
--primary: oklch(0.21 0.034 264.665);
--primary-foreground: oklch(0.985 0.002 247.839);
--secondary: oklch(0.967 0.003 264.542);
--secondary-foreground: oklch(0.21 0.034 264.665);
--muted: oklch(0.967 0.003 264.542);
--muted-foreground: oklch(0.551 0.027 264.364);
--accent: oklch(0.967 0.003 264.542);
--accent-foreground: oklch(0.21 0.034 264.665);
--destructive: oklch(0.577 0.245 27.325);
--border: oklch(0.928 0.006 264.531);
--input: oklch(0.928 0.006 264.531);
--ring: oklch(0.707 0.022 261.325);
--radius: 0.625rem;
--sidebar: oklch(0.985 0.002 247.839);
--sidebar-foreground: oklch(0.13 0.028 261.692);
--sidebar-primary: oklch(0.21 0.034 264.665);
--sidebar-primary-foreground: oklch(0.985 0.002 247.839);
--sidebar-accent: oklch(0.967 0.003 264.542);
--sidebar-accent-foreground: oklch(0.21 0.034 264.665);
--sidebar-border: oklch(0.928 0.006 264.531);
--sidebar-ring: oklch(0.707 0.022 261.325);
}

:root.dark {
color-scheme: dark;

--background: oklch(0.13 0.028 261.692);
--foreground: oklch(0.985 0.002 247.839);
--card: oklch(0.21 0.034 264.665);
--card-foreground: oklch(0.985 0.002 247.839);
--popover: oklch(0.21 0.034 264.665);
--popover-foreground: oklch(0.985 0.002 247.839);
--primary: oklch(0.928 0.006 264.531);
--primary-foreground: oklch(0.21 0.034 264.665);
--secondary: oklch(0.275 0.03 264.364);
--secondary-foreground: oklch(0.985 0.002 247.839);
--muted: oklch(0.275 0.03 264.364);
--muted-foreground: oklch(0.707 0.022 261.325);
--accent: oklch(0.275 0.03 264.364);
--accent-foreground: oklch(0.985 0.002 247.839);
--destructive: oklch(0.704 0.191 22.216);
--border: oklch(1 0 0 / 10%);
--input: oklch(1 0 0 / 15%);
--ring: oklch(0.551 0.027 264.364);
--sidebar: oklch(0.21 0.034 264.665);
--sidebar-foreground: oklch(0.985 0.002 247.839);
--sidebar-primary: oklch(0.488 0.243 264.376);
--sidebar-primary-foreground: oklch(0.985 0.002 247.839);
--sidebar-accent: oklch(0.275 0.03 264.364);
--sidebar-accent-foreground: oklch(0.985 0.002 247.839);
--sidebar-border: oklch(1 0 0 / 10%);
--sidebar-ring: oklch(0.551 0.027 264.364);
}

@layer base {
* {
@apply border-border outline-ring/50;
}
body {
@apply bg-background text-foreground;
}
}"
`;

exports[`Theme generator should add the neutral theme styles to the global stylesheet 1`] = `
"@import "@angular/cdk/overlay-prebuilt.css";

@import "tailwindcss";
@import "@spartan-ng/brain/hlm-tailwind-preset.css";

:root {
color-scheme: light;

--font-sans: '';

--background: oklch(1 0 0);
--foreground: oklch(0.145 0 0);
--card: oklch(1 0 0);
--card-foreground: oklch(0.145 0 0);
--popover: oklch(1 0 0);
--popover-foreground: oklch(0.145 0 0);
--primary: oklch(0.205 0 0);
--primary-foreground: oklch(0.985 0 0);
--secondary: oklch(0.97 0 0);
--secondary-foreground: oklch(0.205 0 0);
--muted: oklch(0.97 0 0);
--muted-foreground: oklch(0.556 0 0);
--accent: oklch(0.97 0 0);
--accent-foreground: oklch(0.205 0 0);
--destructive: oklch(0.577 0.245 27.325);
--border: oklch(0.922 0 0);
--input: oklch(0.922 0 0);
--ring: oklch(0.708 0 0);
--radius: 0.625rem;
--sidebar: oklch(0.985 0 0);
--sidebar-foreground: oklch(0.145 0 0);
--sidebar-primary: oklch(0.205 0 0);
--sidebar-primary-foreground: oklch(0.985 0 0);
--sidebar-accent: oklch(0.97 0 0);
--sidebar-accent-foreground: oklch(0.205 0 0);
--sidebar-border: oklch(0.922 0 0);
--sidebar-ring: oklch(0.708 0 0);
}

:root.dark {
color-scheme: dark;

--background: oklch(0.145 0 0);
--foreground: oklch(0.985 0 0);
--card: oklch(0.205 0 0);
--card-foreground: oklch(0.985 0 0);
--popover: oklch(0.205 0 0);
--popover-foreground: oklch(0.985 0 0);
--primary: oklch(0.922 0 0);
--primary-foreground: oklch(0.205 0 0);
--secondary: oklch(0.269 0 0);
--secondary-foreground: oklch(0.985 0 0);
--muted: oklch(0.269 0 0);
--muted-foreground: oklch(0.708 0 0);
--accent: oklch(0.269 0 0);
--accent-foreground: oklch(0.985 0 0);
--destructive: oklch(0.704 0.191 22.216);
--border: oklch(1 0 0 / 10%);
--input: oklch(1 0 0 / 15%);
--ring: oklch(0.556 0 0);
--sidebar: oklch(0.205 0 0);
--sidebar-foreground: oklch(0.985 0 0);
--sidebar-primary: oklch(0.488 0.243 264.376);
--sidebar-primary-foreground: oklch(0.985 0 0);
--sidebar-accent: oklch(0.269 0 0);
--sidebar-accent-foreground: oklch(0.985 0 0);
--sidebar-border: oklch(1 0 0 / 10%);
--sidebar-ring: oklch(0.556 0 0);
}

@layer base {
* {
@apply border-border outline-ring/50;
}
body {
@apply bg-background text-foreground;
}
}"
`;

exports[`Theme generator should add the slate theme styles to the global stylesheet 1`] = `
"@import "@angular/cdk/overlay-prebuilt.css";

@import "tailwindcss";
@import "@spartan-ng/brain/hlm-tailwind-preset.css";

:root {
color-scheme: light;

--font-sans: '';

--background: oklch(1 0 0);
--foreground: oklch(0.135 0.031 257.417);
--card: oklch(1 0 0);
--card-foreground: oklch(0.135 0.031 257.417);
--popover: oklch(1 0 0);
--popover-foreground: oklch(0.135 0.031 257.417);
--primary: oklch(0.218 0.034 257.417);
--primary-foreground: oklch(0.985 0.003 247.839);
--secondary: oklch(0.967 0.003 257.42);
--secondary-foreground: oklch(0.218 0.034 257.417);
--muted: oklch(0.967 0.003 257.42);
--muted-foreground: oklch(0.554 0.027 257.417);
--accent: oklch(0.967 0.003 257.42);
--accent-foreground: oklch(0.218 0.034 257.417);
--destructive: oklch(0.577 0.245 27.325);
--border: oklch(0.928 0.006 257.419);
--input: oklch(0.928 0.006 257.419);
--ring: oklch(0.707 0.022 257.417);
--radius: 0.625rem;
--sidebar: oklch(0.985 0.003 247.839);
--sidebar-foreground: oklch(0.135 0.031 257.417);
--sidebar-primary: oklch(0.218 0.034 257.417);
--sidebar-primary-foreground: oklch(0.985 0.003 247.839);
--sidebar-accent: oklch(0.967 0.003 257.42);
--sidebar-accent-foreground: oklch(0.218 0.034 257.417);
--sidebar-border: oklch(0.928 0.006 257.419);
--sidebar-ring: oklch(0.707 0.022 257.417);
}

:root.dark {
color-scheme: dark;

--background: oklch(0.135 0.031 257.417);
--foreground: oklch(0.985 0.003 247.839);
--card: oklch(0.218 0.034 257.417);
--card-foreground: oklch(0.985 0.003 247.839);
--popover: oklch(0.218 0.034 257.417);
--popover-foreground: oklch(0.985 0.003 247.839);
--primary: oklch(0.928 0.006 257.419);
--primary-foreground: oklch(0.218 0.034 257.417);
--secondary: oklch(0.275 0.03 257.417);
--secondary-foreground: oklch(0.985 0.003 247.839);
--muted: oklch(0.275 0.03 257.417);
--muted-foreground: oklch(0.707 0.022 257.417);
--accent: oklch(0.275 0.03 257.417);
--accent-foreground: oklch(0.985 0.003 247.839);
--destructive: oklch(0.704 0.191 22.216);
--border: oklch(1 0 0 / 10%);
--input: oklch(1 0 0 / 15%);
--ring: oklch(0.554 0.027 257.417);
--sidebar: oklch(0.218 0.034 257.417);
--sidebar-foreground: oklch(0.985 0.003 247.839);
--sidebar-primary: oklch(0.488 0.243 264.376);
--sidebar-primary-foreground: oklch(0.985 0.003 247.839);
--sidebar-accent: oklch(0.275 0.03 257.417);
--sidebar-accent-foreground: oklch(0.985 0.003 247.839);
--sidebar-border: oklch(1 0 0 / 10%);
--sidebar-ring: oklch(0.554 0.027 257.417);
}

@layer base {
* {
@apply border-border outline-ring/50;
}
body {
@apply bg-background text-foreground;
}
}"
`;

exports[`Theme generator should add the stone theme styles to the global stylesheet 1`] = `
"@import "@angular/cdk/overlay-prebuilt.css";

@import "tailwindcss";
@import "@spartan-ng/brain/hlm-tailwind-preset.css";

:root {
color-scheme: light;

--font-sans: '';

--background: oklch(1 0 0);
--foreground: oklch(0.147 0.004 49.25);
--card: oklch(1 0 0);
--card-foreground: oklch(0.147 0.004 49.25);
--popover: oklch(1 0 0);
--popover-foreground: oklch(0.147 0.004 49.25);
--primary: oklch(0.216 0.006 56.043);
--primary-foreground: oklch(0.985 0.001 106.423);
--secondary: oklch(0.97 0.001 106.424);
--secondary-foreground: oklch(0.216 0.006 56.043);
--muted: oklch(0.97 0.001 106.424);
--muted-foreground: oklch(0.553 0.013 58.071);
--accent: oklch(0.97 0.001 106.424);
--accent-foreground: oklch(0.216 0.006 56.043);
--destructive: oklch(0.577 0.245 27.325);
--border: oklch(0.923 0.003 48.717);
--input: oklch(0.923 0.003 48.717);
--ring: oklch(0.709 0.01 56.259);
--radius: 0.625rem;
--sidebar: oklch(0.985 0.001 106.423);
--sidebar-foreground: oklch(0.147 0.004 49.25);
--sidebar-primary: oklch(0.216 0.006 56.043);
--sidebar-primary-foreground: oklch(0.985 0.001 106.423);
--sidebar-accent: oklch(0.97 0.001 106.424);
--sidebar-accent-foreground: oklch(0.216 0.006 56.043);
--sidebar-border: oklch(0.923 0.003 48.717);
--sidebar-ring: oklch(0.709 0.01 56.259);
}

:root.dark {
color-scheme: dark;

--background: oklch(0.147 0.004 49.25);
--foreground: oklch(0.985 0.001 106.423);
--card: oklch(0.216 0.006 56.043);
--card-foreground: oklch(0.985 0.001 106.423);
--popover: oklch(0.216 0.006 56.043);
--popover-foreground: oklch(0.985 0.001 106.423);
--primary: oklch(0.923 0.003 48.717);
--primary-foreground: oklch(0.216 0.006 56.043);
--secondary: oklch(0.268 0.007 34.298);
--secondary-foreground: oklch(0.985 0.001 106.423);
--muted: oklch(0.268 0.007 34.298);
--muted-foreground: oklch(0.709 0.01 56.259);
--accent: oklch(0.268 0.007 34.298);
--accent-foreground: oklch(0.985 0.001 106.423);
--destructive: oklch(0.704 0.191 22.216);
--border: oklch(1 0 0 / 10%);
--input: oklch(1 0 0 / 15%);
--ring: oklch(0.553 0.013 58.071);
--sidebar: oklch(0.216 0.006 56.043);
--sidebar-foreground: oklch(0.985 0.001 106.423);
--sidebar-primary: oklch(0.488 0.243 264.376);
--sidebar-primary-foreground: oklch(0.985 0.001 106.423);
--sidebar-accent: oklch(0.268 0.007 34.298);
--sidebar-accent-foreground: oklch(0.985 0.001 106.423);
--sidebar-border: oklch(1 0 0 / 10%);
--sidebar-ring: oklch(0.553 0.013 58.071);
}

@layer base {
* {
@apply border-border outline-ring/50;
}
body {
@apply bg-background text-foreground;
}
}"
`;

exports[`Theme generator should add the zinc theme styles to the global stylesheet 1`] = `
"@import "@angular/cdk/overlay-prebuilt.css";

@import "tailwindcss";
@import "@spartan-ng/brain/hlm-tailwind-preset.css";

:root {
color-scheme: light;

--font-sans: '';

--background: oklch(1 0 0);
--foreground: oklch(0.141 0.005 285.823);
--card: oklch(1 0 0);
--card-foreground: oklch(0.141 0.005 285.823);
--popover: oklch(1 0 0);
--popover-foreground: oklch(0.141 0.005 285.823);
--primary: oklch(0.21 0.006 285.885);
--primary-foreground: oklch(0.985 0 0);
--secondary: oklch(0.967 0.001 286.375);
--secondary-foreground: oklch(0.21 0.006 285.885);
--muted: oklch(0.967 0.001 286.375);
--muted-foreground: oklch(0.552 0.016 285.938);
--accent: oklch(0.967 0.001 286.375);
--accent-foreground: oklch(0.21 0.006 285.885);
--destructive: oklch(0.577 0.245 27.325);
--border: oklch(0.92 0.004 286.32);
--input: oklch(0.92 0.004 286.32);
--ring: oklch(0.705 0.015 286.067);
--radius: 0.625rem;
--sidebar: oklch(0.985 0 0);
--sidebar-foreground: oklch(0.141 0.005 285.823);
--sidebar-primary: oklch(0.21 0.006 285.885);
--sidebar-primary-foreground: oklch(0.985 0 0);
--sidebar-accent: oklch(0.967 0.001 286.375);
--sidebar-accent-foreground: oklch(0.21 0.006 285.885);
--sidebar-border: oklch(0.92 0.004 286.32);
--sidebar-ring: oklch(0.705 0.015 286.067);
}

:root.dark {
color-scheme: dark;

--background: oklch(0.141 0.005 285.823);
--foreground: oklch(0.985 0 0);
--card: oklch(0.21 0.006 285.885);
--card-foreground: oklch(0.985 0 0);
--popover: oklch(0.21 0.006 285.885);
--popover-foreground: oklch(0.985 0 0);
--primary: oklch(0.92 0.004 286.32);
--primary-foreground: oklch(0.21 0.006 285.885);
--secondary: oklch(0.274 0.006 286.033);
--secondary-foreground: oklch(0.985 0 0);
--muted: oklch(0.274 0.006 286.033);
--muted-foreground: oklch(0.705 0.015 286.067);
--accent: oklch(0.274 0.006 286.033);
--accent-foreground: oklch(0.985 0 0);
--destructive: oklch(0.704 0.191 22.216);
--border: oklch(1 0 0 / 10%);
--input: oklch(1 0 0 / 15%);
--ring: oklch(0.552 0.016 285.938);
--sidebar: oklch(0.21 0.006 285.885);
--sidebar-foreground: oklch(0.985 0 0);
--sidebar-primary: oklch(0.488 0.243 264.376);
--sidebar-primary-foreground: oklch(0.985 0 0);
--sidebar-accent: oklch(0.274 0.006 286.033);
--sidebar-accent-foreground: oklch(0.985 0 0);
--sidebar-border: oklch(1 0 0 / 10%);
--sidebar-ring: oklch(0.552 0.016 285.938);
}

@layer base {
* {
@apply border-border outline-ring/50;
}
body {
@apply bg-background text-foreground;
}
}"
`;
